### PR TITLE
Support S3 File Field Upload functionality

### PR DESCRIPTION
**Pull request by eagw**
Update the client to pass additional information to multinetjs to allow for table and network uploads using the S3 File Field client.

This PR works in conjunction with the s3ff branch of [multinetjs](https://github.com/multinet-app/multinetjs/tree/s3ff)

Should be merged after #178.  Commits pertinent to this PR are 9e1d1df, 86dd4fc and fcaec43, which make changes to FileUploadForm.vue and TableDialog.vue (all other commits are for changes addressed in #178).
**Edit: Now that #178 is merged, all commits associated with this PR are relevant.**

### CODE REVIEW

File: src/components/TableDialog.vue
Comment (waxlamp) [inline src/components/TableDialog.vue:224]: There's a better way to do this. Extract the object from `sampleRows` as you are doing in line 223, then call `.hasOwnProperty()` on it to detect the `_from` and `_to` attributes.

You may run into a linter error when you do this--this [documentation](https://eslint.org/docs/rules/no-prototype-builtins) explains what's going on and how to fix it.

File: src/components/FileUploadForm.vue
Comment (waxlamp) [inline src/components/FileUploadForm.vue:187]: If my [comment](https://github.com/multinet-app/multinetjs/pull/44#pullrequestreview-728659778) on the corresponding multinetjs PR is correct, then this will need to change.

**Review by waxlamp — CHANGES_REQUESTED** [review]

File: src/components/TableDialog.vue
Comment (AlmightyYakob) [inline src/components/TableDialog.vue:308]: Same as roni's comment about the corresponding change in multinetJS

Comment (eagw) [reply src/components/TableDialog.vue]: See 5f46644

File: src/components/TableDialog.vue
Comment (waxlamp) [inline src/components/TableDialog.vue:221]: For readability, you may want to split this up into a couple of `const` expressions, like:

```typescript
const hasFrom = Object.prototype.hasOwnProperty.call(sampleRows.value[0]);
```

In general, when you have complex expressions composed into a long line of code, it's good to consider the structure of it for purposes of readability to other devs (including Emi-from-the-future).

I think you also want to extract the `keys` value as in your old code:

```typescript
const sample = sampleRows.value[0] || {};
```

so that you aren't operating on `undefined` by accident. Essentially, you want the type of `sample` to be `Node` rather than `Node | undefined` (I'm saying `Node` because I forgot the true name of the type, but my point stands).

File: src/components/TableDialog.vue
Comment (eagw) [reply src/components/TableDialog.vue]: That's helpful, thank you!

See 701bec8

File: src/components/TableDialog.vue
Comment (eagw) [reply src/components/TableDialog.vue]: Thank you for the feedback, those are both great points!

See a970386

File: src/components/FileUploadForm.vue
Comment (AlmightyYakob) [inline src/components/FileUploadForm.vue:183]: Why was this changed to `uploadNetwork`?

**Review by waxlamp — COMMENTED** [review]
LGTM, pending Jake's approval.

File: src/components/FileUploadForm.vue
Comment (eagw) [reply src/components/FileUploadForm.vue:183]: This is the component that handles uploading files to create networks, so I switched it to use the new separate function.  TableDialog.vue takes care of uploading files to make tables.

**Review by AlmightyYakob — APPROVED** [review]

**Pull request merged**
